pkg/appmanager/resourceConfig.go

```diff
@@ -2071,25 +2071,57 @@ func (appMgr *Manager) handleRouteTls(
 		rc.MergeRules(appMgr.mergedRulesMap)
 	}
 
-	// Add whitelist conditions to every rule (to prevent access)
+	// Add whitelist condition
 	var whitelistSourceRanges []string
 	if sourceRange, ok := route.ObjectMeta.Annotations[f5VsWhitelistSourceRangeAnnotation]; ok {
 		whitelistSourceRanges = parseWhitelistSourceRangeAnnotations(sourceRange)
 	}
 	if len(whitelistSourceRanges) > 0 {
 		for _, pol := range rc.Policies {
 			if pol.Name == policyName {
-				for _, rl := range pol.Rules {
-					cond := condition{
-						Tcp:     true,
-						Address: true,
-						Matches: true,
-						Name:    "0",
-						Values:  whitelistSourceRanges,
+				for i, rl := range pol.Rules {
+					if rl.FullURI == rule.FullURI && !strings.HasSuffix(rl.Name, "-reset") {
```

Comment:
Is this run before or after sorting?

If before, do we always get the correct ordering, I have a test that will sometimes reorder equal strings.

If after, should we be updating ordinals on the previously set rules?

Reply:
This is run before sorting.

Reply:
I'm worried we might move these out of order, if you run this: https://play.golang.org/p/7bRNfjF4SRF you'll see the original ordering doesn't always get preserved. Moving the original position of the quux.com strings effects the final position.